src/AssetsController.php

```diff
@@ -0,0 +1,151 @@
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+	}
+
+	/**
+	 * Register block scripts & styles.
+	 *
+	 * @since 2.5.0
+	 * Moved data related enqueuing to new AssetDataRegistry class as part of ongoing refactoring.
+	 */
+	public function register_assets() {
+		$this->register_style( 'wc-block-editor', plugins_url( $this->api->get_block_asset_build_path( 'editor', 'css' ), __DIR__ ), array( 'wp-edit-blocks' ) );
+		$this->register_style( 'wc-block-style', plugins_url( $this->api->get_block_asset_build_path( 'style', 'css' ), __DIR__ ), array( 'wc-block-vendors-style' ) );
+
+		wp_style_add_data( 'wc-block-editor', 'rtl', 'replace' );
+		wp_style_add_data( 'wc-block-style', 'rtl', 'replace' );
+
+		$this->api->register_script( 'wc-blocks-middleware', 'build/wc-blocks-middleware.js', [], false );
+		$this->api->register_script( 'wc-blocks-data-store', 'build/wc-blocks-data.js', [ 'wc-blocks-middleware' ] );
+		$this->api->register_script( 'wc-blocks', $this->api->get_block_asset_build_path( 'blocks' ), [], false );
+		$this->api->register_script( 'wc-vendors', $this->api->get_block_asset_build_path( 'vendors' ), [], false );
+		$this->api->register_script( 'wc-blocks-registry', 'build/wc-blocks-registry.js', [], false );
+		$this->api->register_script( 'wc-shared-context', 'build/wc-shared-context.js', [] );
+		$this->api->register_script( 'wc-shared-hocs', 'build/wc-shared-hocs.js', [], false );
+		$this->api->register_script( 'wc-price-format', 'build/price-format.js', [], false );
+
+		if ( Package::feature()->is_feature_plugin_build() ) {
+			$this->api->register_script( 'wc-blocks-checkout', 'build/blocks-checkout.js', [] );
+		}
+
+		wp_add_inline_script(
+			'wc-blocks-middleware',
+			"
+			var wcBlocksMiddlewareConfig = {
+				storeApiNonce: '" . esc_js( wp_create_nonce( 'wc_store_api' ) ) . "',
+				wcStoreApiNonceTimestamp: '" . esc_js( time() ) . "'
+			};
+			",
+			'before'
+		);
+	}
+
+	/**
+	 * Register the vendors style file. We need to do it after the other files
+	 * because we need to check if `wp-edit-post` has been enqueued.
+	 */
+	public function enqueue_scripts() {
+		// @todo Remove fix to load our stylesheets after editor CSS.
+		// See #3068 and #3898 for the rationale of this fix. It should be no
+		// longer necessary when the editor is loaded in an iframe (https://github.com/WordPress/gutenberg/issues/20797).
+		$this->register_style( 'wc-block-vendors-style', plugins_url( $this->api->get_block_asset_build_path( 'vendors-style', 'css' ), __DIR__ ), wp_style_is( 'wp-edit-post' ) ? [ 'wp-edit-post' ] : [] );
+	}
+
+	/**
+	 * Add body classes to the frontend and within admin.
+	 *
+	 * @param string|array $classes Array or string of CSS classnames.
+	 * @return string|array Modified classnames.
+	 */
+	public function add_theme_body_class( $classes ) {
```

Comment:
Note I combined the 2 functions into one which handles $classes based on type instead of 2 functions for similar logic.